src/components/main-view/main-view.jsx

```diff
@@ -47,94 +54,123 @@ export const MainView = () => {
 
         setMovies(moviesFromApi);
       });
-
-    // .catch((error) => console.error("Error fetching movies:", error));
   }, [token]);
 
-  if (selectedMovie) {
-    let similarMovies = movies.filter(
-      (movie) => movie.Genre.Name === selectedMovie.Genre.Name
-    );
-    similarMovies = similarMovies.filter(
-      (movie) => movie._id !== selectedMovie._id
-    );
-    similarMovies = similarMovies.slice(0, 4);
-
-    return (
-      <>
-        <MovieView
-          movie={selectedMovie}
-          onBackClick={() => setSelectedMovie(null)}
+  return (
+    <BrowserRouter>
+      <Container>
+        <NavigationBar
+          user={user}
+          onLoggedOut={() => {
+            setUser(null);
+            setToken(null);
+            localStorage.clear();
+          }}
         />
-        <hr />
-        <h2>Similar movies</h2>
-        <Row>
-          {similarMovies.map((movie) => (
-            <Col className="mb-5" key={movie._id} md={3}>
-              <MovieCard
-                movie={movie}
-                onMovieClick={(newSelectedMovie) => {
-                  setSelectedMovie(newSelectedMovie);
-                }}
-              />
-            </Col>
-          ))}
-        </Row>
-      </>
-    );
-  }
 
-  return (
-    <Row className="justify-content-md-center">
-      {!user ? (
-        <>
-          <Col md={5}>
-            <LoginView
-              onLoggedIn={(user, token) => {
-                setUser(user);
-                setToken(token);
-              }}
+        <Row className="justify-content-md-center">
+          <Routes>
+            <Route
+              path="/signup"
+              element={
+                <>
+                  {user ? (
+                    <Navigate to="/" />
+                  ) : (
+                    <Col md={5}>
+                      <SignupView
+                        onSignup={(user, token) => {
+                          setUser(user);
+                          setToken(token);
+                        }}
+                      />
+                    </Col>
+                  )}
+                </>
+              }
+            />
+
+            <Route
+              path="/profile"
```

Comment:
I would make these constants and move them to a separate "routes" file then import them. Having to chase down all the strings in multiple files during a refactor will slow you down big time. Changing them one place =  🧠